app/controllers/DetailedIncomeTaxEstimateController.scala

```diff
@@ -63,7 +63,7 @@ class DetailedIncomeTaxEstimateController @Inject()(
     } yield {
       (totalTax, maybeTaxCodeIncomes, taxSummary, nonTaxCode) match {
         case (
-            TaiSuccessResponseWithPayload(totalTax: TotalTax),
+            totalTax,
```

Comment:
I would suggest using mapN as we did elsewhere: 
```
def taxExplanationPage(): Action[AnyContent] = (authenticate andThen validatePerson).async { implicit request =>
    val nino = request.taiUser.nino
    implicit val user: AuthedUser = request.taiUser
    (
      taxAccountService.totalTax(nino, TaxYear()),
      taxAccountService.taxCodeIncomes(nino, TaxYear()),
      taxAccountService.taxAccountSummary(nino, TaxYear()),
      codingComponentService.taxFreeAmountComponents(nino, TaxYear()),
      taxAccountService.nonTaxCodeIncomes(nino, TaxYear())
      ).mapN {
      case (
        totalTax, Right(taxCodeIncomes),
        TaiSuccessResponseWithPayload(taxAccountSummary: TaxAccountSummary),
        codingComponents,
        TaiSuccessResponseWithPayload(nonTaxCodeIncome: NonTaxCodeIncome)) =>
        val model = DetailedIncomeTaxEstimateViewModel(
          totalTax,
          taxCodeIncomes,
          taxAccountSummary,
          codingComponents,
          nonTaxCodeIncome)
        Ok(detailedIncomeTaxEstimate(model))
      case _ =>
        errorPagesHandler.internalServerError("Failed to fetch total tax details")
    }.recover {
      case NonFatal(e) => errorPagesHandler.internalServerError("Failed to fetch total tax details", Some(e))
    }
  }
  ```

Reply:
You could even skip the: 
`case _ =>
        errorPagesHandler.internalServerError("Failed to fetch total tax details")`
As it would go to the recover in case the pattern match fails